### ISSUE TITLE
False positive E0015 for const parameters

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#5635]
## Environment

* **IntelliJ Rust plugin version:** 0.2.125.3167-202-nightly
* **Rust toolchain version:** 1.46.0-nightly (a8cf39911 2020-06-21) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2020.2 EAP Ultimate Edition (IU-202.5792.28)
* **Operating system:** macOS 10.13.6

## Problem description
![image](https://user-images.githubusercontent.com/2539310/85320767-45775d00-b4cc-11ea-8b25-c81fb7ed6194.png)

Was introduced in #5436

## Steps to reproduce
```rust
fn new_array<const N: usize>() -> [u8; N] {
    [0; N]
}
```
